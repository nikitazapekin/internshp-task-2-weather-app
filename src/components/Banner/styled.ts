import styled from "styled-components";

export const Wrapper = styled.section`
  max-width: ${({ theme }) => theme.container}px;
  width: 100%;
  position: relative;
  backdrop-filter: blur(5px);
  padding: ${({ theme }) => theme.spaces.sm}px ${({ theme }) => theme.spaces.md}px
    ${({ theme }) => theme.spaces.md}px ${({ theme }) => theme.spaces.md}px;
`;

export const BannerBackground = styled.section`
  background-color: ${({ theme }) => theme.colors.gray};
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: ${({ theme }) => theme.zIndexes.x};
  opacity: 0.2;
`;
