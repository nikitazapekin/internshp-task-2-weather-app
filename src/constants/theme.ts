export const theme = {
  container: 1187,
  deviceHeight: "100vh",
  colors: {
    black: "#000",
    blue: "#122e65",
    gray: "#D9D9D9",
    white: "#fff",
  },
  fontFamilies: {
    primary: "Inter",
    secondary: "sans-serif",
  },
  fontSizes: {
    xl: "41px",
    lg: "30px",
    md: "27px",
    sm: "22px",
    xs: "20px",
    xxs: "18px",
  },
  fontWeights: {
    regular: 400,
  },
  media: {
    xxl: 1400,
    xl: 1280,
    lg: 1024,
    md: 768,
    xh: 560,
    sm: 480,
  },
  zIndexes: {
    x: -1,
    xs: 0,
    sm: 1,
    md: 2,
  },
  spaces: {
    x: 5,
    xs: 10,
    sm: 15,
    md: 20,
    lg: 30,
    xl: 40,
    xxl: 50,
    g: 60,
    h: 70,
    xh: 80,
    eh: 90,
  },
  swiper: {
    item_gap: 5,
    margin_base: 2.5,
    vw_ratio: 0.5,
    scroll_snap_type: "x mandatory",
    scroll_snap_align: "start",
    flex_basis: "0 0",
    min_width: "0",
    container: 16.6,
  },
};
