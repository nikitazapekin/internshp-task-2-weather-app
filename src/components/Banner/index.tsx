import BottomOfTheBanner from "@components/BottomOfTheBanner";
import TopOfTheBanner from "@components/TopOfTheBanner";

import { BannerBackground, Wrapper } from "./styled";

const Banner = () => {
  return (
    <Wrapper>
      <TopOfTheBanner />

      <BottomOfTheBanner />
      <BannerBackground />
    </Wrapper>
  );
};

export default Banner;
