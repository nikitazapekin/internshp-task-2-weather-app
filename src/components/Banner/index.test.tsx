import { theme } from "@constants/theme";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import Banner from ".";

jest.mock("@components/BottomOfTheBanner", () => () => <div data-testid="bottom-banner" />);
jest.mock("@components/CenterOfBanner", () => () => <div data-testid="center-banner" />);
jest.mock("@components/TopOfTheBanner", () => () => <div data-testid="top-banner" />);
describe("Banner Component", () => {
  test("matches snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Banner />
      </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
