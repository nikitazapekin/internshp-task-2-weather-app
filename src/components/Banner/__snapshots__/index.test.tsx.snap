// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Banner Component matches snapshot 1`] = `
<section
  class="sc-bRKDuR knghSf"
  data-testid="banner"
>
  <div
    class="sc-fhHczv jHoGvg"
  >
    <div
      data-testid="top-banner"
    />
    <div
      data-testid="center-banner"
    />
  </div>
  <div
    data-testid="bottom-banner"
  />
  <div
    class="sc-hvigdm eKwmgj"
  />
</section>
`;
